### PR TITLE
Check that we don't write -- in markdown

### PR DESCRIPTION
**Pull request by mandolaerik**
- Check that we don't write -- in markdown

### CODE REVIEW

**Comment by syssimics** [comment]
Verification [#10575984](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10575984): fail

**Comment by lwaern-intel** [comment]
heh. the `--` found in verification is also my fault.

File: validate_md_links.py [hunk 1]
```diff
@@ -95,6 +95,11 @@ def main():
                 line = md_files[path][:match.start()].count('\n') + 1
                 sys.stderr.write(f'{path}:{line}: error: {message}\n')
                 ok = False
+        mdash = md_files[path].find(' -- ')
```
Comment (lwaern-intel) [inline validate_md_links.py:98]: Consider also lines beginning or ending with `--`.

Comment (mandolaerik) [reply validate_md_links.py]: done

**Comment by syssimics** [comment]
Verification [#10586886](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10586886): pass

File: validate_md_links.py [hunk 2]
```diff
@@ -95,6 +95,11 @@ def main():
                 line = md_files[path][:match.start()].count('\n') + 1
                 sys.stderr.write(f'{path}:{line}: error: {message}\n')
                 ok = False
+        for (i, line) in enumerate(md_files[path].split('\n')):
+            if ' -- ' in f' {line} ':
+                sys.stderr.write(
+                    f'{path}:{i+1}: error: replace -- with &mdash;\n')
+                ok = False
     sys.exit(0 if ok else 1)
```
Comment (lwaern-intel) [inline validate_md_links.py:103]: `sys.exit(1 * ok)`! _Embrace_ the weakly-typed heresy!

<sub><sub>do not actually do this</sub></sub>

**Review by lwaern-intel — APPROVED** [review]

File: validate_md_links.py (hunk 2 above)
Comment (mandolaerik) [reply validate_md_links.py:103]: that would be `1-ok`, right?

One alternative representation that actually would make a bit of sense, is to go all-in on integers and rename `ok` to `exitcode`.

**Pull request merged**